Antenna driver

zaq32 opened the pull request:
First draft of the driver for antenna deployment system.

<!-- Reviewable:start -->

---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pw-sat2/pwsat2obc/40)
<!-- Reviewable:end -->

zaq32 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[Driver] Anteny](https://trello.com/c/k4XSdbQW/70-driver-anteny)

Novakov commented:
Reviewed 11 of 34 files at r1, 5 of 13 files at r2, 15 of 20 files at r3, 10 of 10 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 27 unresolved discussions.

---

*[integration_tests/devices/antenna.py, line 40 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSQE2mmVVGgCpewdO6:-KWSQE2mmVVGgCpewdO7:b-9xpfkg) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L40)):*
> ```Python
>     on_get_temperature = None
>     on_get_deployment_status = None
>     on_deployment_cancel = None
> ```

As these fields are callbacks, it would be nice to have signatures here (comments are fine)

---

*[integration_tests/devices/antenna.py, line 44 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSXHVp1TwKSz0LSmgk:-KWSXHVp1TwKSz0LSmgl:brvzy17) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L44)):*
> ```Python
>     def __init__(self, address):
>         super(AntennaController, self).__init__(address)
>         self.log = logging.getLogger("Antenna")
> ```

Append controller address here (like `Antenna.32`) and don't put it in each log message

---

*[integration_tests/devices/antenna.py, line 66 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSPzX1SXiB27t7vElf:-KWSPzX1SXiB27t7vElg:b2c4h2v) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L66)):*
> ```Python
>                 return result
> 
>         return default
> ```

Move it to utils

---

*[integration_tests/devices/antenna.py, line 86 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSQ_QFOK_x2YQ5jW6g:-KWSQ_QFOK_x2YQ5jW6h:b-skb9wx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L86)):*
> ```Python
>             self.armed = self.call(self.on_arm_state_change, False, False)
> 
>     def deploy_antenna(self, antanna_id, timeout):
> ```

ant**e**nna_id

The same typo in many methods ;)

---

*[integration_tests/devices/antenna.py, line 172 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSXnl-6g1Ts6f-_33H:-KWSXnl-6g1Ts6f-_33I:brndtix) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L172)):*
> ```Python
>         result = self.call(self.on_get_temperature, None)
>         if result is None:
>             result = 0
> ```

Coallesce by `self.call(...) or 0` 

---

*[integration_tests/devices/antenna.py, line 174 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSUCvyYMnhgKtN5PbE:-KWSUCvyYMnhgKtN5PbF:b-3ktviu) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L174)):*
> ```Python
>             result = 0
> 
>         return [(result >> 8) & 0xff, result & 0xff]
> ```

Python have standard functions for packing values:

`struct.pack('>h', result)`

Docs: https://docs.python.org/2/library/struct.html

---

*[integration_tests/devices/antenna.py, line 200 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSVeVUaz3wqAfiSMNq:-KWSVeVUaz3wqAfiSMNr:b-s620lq) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L200)):*
> ```Python
>         lsb = update_value(lsb, self.deployment_in_progress, 0x10)
>         lsb = update_value(lsb, self.armed, 0x1)
>         return [msb, lsb]
> ```

You can use `reduce(lambda s,b:  (s << 1) | b, array_of_bools, 0)` to convert array of bools to int, then pack it to two bytes.

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWSW2sMB35JJVbg6Qj4:b-9u86pd) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
> ```Python
> 
>         self.ignore_deployment_switch = False
>         self.log.debug("Antenna deployment finished on controller: %d", self.address)
> ```

Each device has two interfaces: API for tests and methods called via I2C. I think it would be beneficial to separete these two interfaces, for example by preceding them with `_` as they are not intended to be invoked by tests

Second thing, how about putting comment with each command with paragraph from ICD (if possible)? It will be easier to find where each command is described

---

*[integration_tests/obc/antenna.py, line 72 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSOvm9Wii9fdZTszke:-KWSOvm9Wii9fdZTszkf:b57be5q) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/obc/antenna.py#L72)):*
> ```Python
> 
>     def parse_deployment_state(result):
>         parts = result.split(" ");
> ```

If all values are int, it is possible to convert them all at once:
`map(int, result.split(" ")`

And no semicolon at the end ;)

Private methods should begin with `_`

---

*[integration_tests/obc/antenna.py, line 80 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSPI2aG0ak-8xd-Y5p:-KWSPI2aG0ak-8xd-Y5q:b-c5tcl6) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/obc/antenna.py#L80)):*
> ```Python
>         parsedResult = AntennaStatus(True)
>         for cx in range(4):
>             parsedResult.DeploymentState[cx] = int(parts[cx + 1])
> ```

If input is converted to int, this loop can be simplified: `parsedResult.DeploymentState = parts[1:5]` 

---

*[integration_tests/system/system.py, line 11 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS5EwTSirPIwwfpNaU:-KWS5EwTSirPIwwfpNaV:b2a7hiw) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/system/system.py#L11)):*
> ```Python
> class System:
>     def __init__(self, obc_com, sys_bus_com, payload_bus_com, use_single_bus, gpio, auto_power_on = True):
>         logging.basicConfig()
> ```

Logging configuration is done in `tests/__init__.py`. Don't do it here.

---

*[integration_tests/system/system.py, line 49 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS6cIeaYs4a9qjeh5E:-KWS6cIeaYs4a9qjeh5F:b-uo8qu2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/system/system.py#L49)):*
> ```Python
>         self.sys_bus.add_device(self.receiver)
>         self.sys_bus.add_device(self.primary_antenna)
>         self.sys_bus.add_device(self.backup_antenna)
> ```

Should backup antenna controller be on payload bus?

---

*[integration_tests/tests/test_antenna.py, line 17 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdTV3ZngOSVcLmom3m:-KWdTV3_5Bp5qAo29m_a:bxecb3q) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L17)):*
> ```Python
> 
>     def wait_for_change(self, timeout = None):
>         return self.flag.wait(timeout)
> ```

Move to utils

---

*[integration_tests/tests/test_antenna.py, line 23 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdTZmz9hUz-gxPRWcG:-KWdTZmz9hUz-gxPRWcH:bse1yjq) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L23)):*
> ```Python
>         result = super(Test_Antenna, self).__init__(methodName)
>         self.auto_power_on = False
>         return result
> ```

I'd prefer to have this as decorator `@auto_power_on(False)`

---

*[integration_tests/tests/test_antenna.py, line 97 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdUCiqasG0Ku3CCKNa:-KWdUCiqasG0Ku3CCKNb:b6ggnma) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L97)):*
> ```Python
>         self.assertFalse(result.DeploymentInProgress[1])
>         self.assertFalse(result.DeploymentInProgress[2])
>         self.assertTrue(result.DeploymentInProgress[3])
> ```

Fon't write assertTrue/False without meaningful message unless you love debugging series of `Test failed. Expected: True Actual: False` (I don't)

---

*[libs/base/reader.c, line 7 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS5-H2k1k5VRTaortO:-KWS5-H2k1k5VRTaortP:b-aqmax3) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/base/reader.c#L7)):*
> ```C
>     if (reader->status)
>     {
>         reader->status = (reader->position + requestedSize) <= reader->length;
> ```

It took me a while to understand what `status` is. I think it will be better to name it `isValid` or `stillInsideBuffer`

---

*[libs/drivers/antenna/antenna.c, line 14 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWScB2QmLfMx2XBRTi6:-KWScB2QmLfMx2XBRTi7:bdf3lbg) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L14)):*
Copy-paste? 
> to be reset

---

*[libs/drivers/antenna/antenna.c, line 113 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSeqWF6WAgqc93wnjS:-KWSeqWF6WAgqc93wnjT:bke0nap) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L113)):*
> ```C
> }
> 
> AntennaDeploymentProcessStatus IsDeploymentActive(struct AntennaDriver* driver, AntennaChannel channel)
> ```

Missing `static`

---

*[libs/drivers/antenna/antenna.c, line 144 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSf58CiSHQXDT51b5F:-KWSf58CiSHQXDT51b5G:bke0nap) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L144)):*
> ```C
> }
> 
> OSResult GetDeploymentStatus(struct AntennaDriver* driver,
> ```

Missing `static`

---

*[libs/drivers/antenna/antenna.c, line 243 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSeIYZz-oGcToWrtCj:-KWSeIY_ECGEicoLreLp:bs8t3rn) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L243)):*
> ```C
>             telemetry->ActivationTime[1] = secondaryValue;
>             telemetry->flags |= flags[i];
>         }
> ```

Why always checking both controllers even if primary is working?

---

*[libs/drivers/antenna/miniport.c, line 22 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS_njzYoYc02Vzxa6B:-KWS_njzYoYc02Vzxa6C:bquyto8) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L22)):*
> ```C
>     QUERY_TEMPERATURE = 0xc0,
>     QUERY_DEPLOYMENT_STATUS = 0xc3
> } Command;
> ```

Why are some enum names CamelCase (like OSResult) but this is upper case with underscores without any prefix?

---

*[libs/drivers/antenna/miniport.c, line 54 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSZlfdoR5ck9b98YK-:-KWSZlfdoR5ck9b98YK0:b3oid8r) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L54)):*
> ```C
> 
>     return status;
> }
> ```

I see some repeating pattern of sending commands without parameters with optional response. Just thinking maybe is useful to extract it to some common utility functions?

---

*[libs/drivers/antenna/miniport.c, line 199 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSaoDy09S8NSwoX4_w:-KWSaoDy09S8NSwoX4_x:b-t3casx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L199)):*
> ```C
>     telemetry->IgnoringDeploymentSwitches = (value & 0x0100) != 0;
>     telemetry->DeploymentSystemArmed = (value & 0x0001) != 0;
>     return OSResultSuccess;
> ```

AVR-Libc defines few useful macros for bit operations: `bit_is_set` and `bit_is_clear`. I would say it will make code more readable if we include these functions in our codebase 

Source: http://www.nongnu.org/avr-libc/user-manual/sfr__defs_8h_source.html#234

---

*[libs/drivers/antenna/Include/antenna/miniport.h, line 21 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWcTLGfIUeI4QPAa3gC:-KWcTLGfIUeI4QPAa3gD:b-64quja) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/Include/antenna/miniport.h#L21)):*
> ```objc
>  *
>  * The driver itself is responsible mainly for building & parsing frames exchanged with
>  * the hardware itself any extensive data interpretation is not done t this level.
> ```

**a**t this level

---

*[libs/drivers/antenna/Include/antenna/miniport.h, line 30 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWcTRvyKrXsTuCTkPW0:-KWcTRvyKrXsTuCTkPW1:bqnfyid) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/Include/antenna/miniport.h#L30)):*
> ```objc
>  * This low level driver is responsible for building/parsing frames exchanges with
>  * hardware. By itself is driver is completely stateless and can be used to manage
>  * multiple hardware channels at the same time (event concurrently). It is the caller
> ```

even**t** concurrently - that's not exaclty true. You can't send two commands at the same time on two channels if they are using the same underlying hardware bus. 

---

*[src/commands/antenna.c, line 63 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWZg0mdJk7TOxMu2y_G:-KWZg0mdJk7TOxMu2y_H:b2c4h2v) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/src/commands/antenna.c#L63)):*
> ```C
> {
>     return value ? 1 : 0;
> }
> ```

Move it to utils

---

*[unit_tests/antenna/miniport.cpp, line 133 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdSHWhx2cw9wJaPgGB:-KWdSHWhx2cw9wJaPgGC:bcpm0tg) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/unit_tests/antenna/miniport.cpp#L133)):*
> ```cpp
>             EXPECT_THAT(length, Eq(2u));
>             EXPECT_THAT(inData[1], Eq(200u));
>             return I2CResultOK;
> ```

(Just thinking) how about creating some matchers dedicated for I2C transfers?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Review status: 32 of 46 files reviewed at latest revision, 27 unresolved discussions.

---

*[integration_tests/devices/antenna.py, line 40 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSQE2mmVVGgCpewdO6:-KWnY4NRBQ_S7WyLKUk4:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L40)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> As these fields are callbacks, it would be nice to have signatures here (comments are fine)
</details>
Fixed

---

*[integration_tests/devices/antenna.py, line 44 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSXHVp1TwKSz0LSmgk:-KWnYvBZk_EoO0sx_eZr:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L44)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Append controller address here (like `Antenna.32`) and don't put it in each log message
</details>
Fixed

---

*[integration_tests/devices/antenna.py, line 66 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSPzX1SXiB27t7vElf:-KWnZF1YeWGLImHpXGC7:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L66)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Move it to utils
</details>
Fixed

---

*[integration_tests/devices/antenna.py, line 86 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSQ_QFOK_x2YQ5jW6g:-KWnZMn_xP2oJeDK9FRS:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L86)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> ant**e**nna_id
> 
> The same typo in many methods ;)
</details>
Fixed

---

*[integration_tests/devices/antenna.py, line 172 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSXnl-6g1Ts6f-_33H:-KWnZx9uIe52ZQeoOM4x:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L172)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Coallesce by `self.call(...) or 0` 
</details>
Fixed

---

*[integration_tests/devices/antenna.py, line 174 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSUCvyYMnhgKtN5PbE:-KWna7Qr0VtJSMX4Bh3N:b-9pjj5j) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L174)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Python have standard functions for packing values:
> 
> `struct.pack('>h', result)`
> 
> Docs: https://docs.python.org/2/library/struct.html
> 
</details>
I am not sure that I can use it and get anything simpler than this that produces byte array

---

*[integration_tests/devices/antenna.py, line 200 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSVeVUaz3wqAfiSMNq:-KWoF_S_yIvTVGRWZlwW:b-lnlbnu) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L200)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> You can use `reduce(lambda s,b:  (s << 1) | b, array_of_bools, 0)` to convert array of bools to int, then pack it to two bytes.
</details>
It may work but I still prefer explicit function call.

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWndISwWsZmeC063fX1:bb6vskf) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Each device has two interfaces: API for tests and methods called via I2C. I think it would be beneficial to separete these two interfaces, for example by preceding them with `_` as they are not intended to be invoked by tests
> 
> Second thing, how about putting comment with each command with paragraph from ICD (if possible)? It will be easier to find where each command is described
</details>
Fine

---

*[integration_tests/obc/antenna.py, line 72 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSOvm9Wii9fdZTszke:-KWndjUP3V_kQG_aPZOE:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/obc/antenna.py#L72)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> If all values are int, it is possible to convert them all at once:
> `map(int, result.split(" ")`
> 
> And no semicolon at the end ;)
> 
> Private methods should begin with `_`
</details>
Fixed

---

*[integration_tests/obc/antenna.py, line 80 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSPI2aG0ak-8xd-Y5p:-KWne237bdCJVISMaWda:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/obc/antenna.py#L80)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> If input is converted to int, this loop can be simplified: `parsedResult.DeploymentState = parts[1:5]` 
</details>
Fixed

---

*[integration_tests/system/system.py, line 11 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS5EwTSirPIwwfpNaU:-KWneJ0Qi1xQe0WF3KTc:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/system/system.py#L11)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Logging configuration is done in `tests/__init__.py`. Don't do it here.
</details>
Fixed

---

*[integration_tests/system/system.py, line 49 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS6cIeaYs4a9qjeh5E:-KWneP5osZtykD5AzetR:b-81la7d) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/system/system.py#L49)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Should backup antenna controller be on payload bus?
</details>
Right

---

*[integration_tests/tests/test_antenna.py, line 17 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdTV3ZngOSVcLmom3m:-KWnhGW1nj72MLY3eh0T:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L17)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Move to utils
</details>
Fixed

---

*[integration_tests/tests/test_antenna.py, line 97 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdUCiqasG0Ku3CCKNa:-KWne_6GaJoG-lHTy2IJ:bv06f12) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L97)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Fon't write assertTrue/False without meaningful message unless you love debugging series of `Test failed. Expected: True Actual: False` (I don't)
</details>
what would you put in such message that is not already expressed in the code?

---

*[libs/base/reader.c, line 7 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS5-H2k1k5VRTaortO:-KWoGIsxTtWLMThJsoPi:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/base/reader.c#L7)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> It took me a while to understand what `status` is. I think it will be better to name it `isValid` or `stillInsideBuffer`
</details>
Fixed

---

*[libs/drivers/antenna/antenna.c, line 14 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWScB2QmLfMx2XBRTi6:-KWoDiIvsnHHOFO_cD6G:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L14)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Copy-paste? 
> > to be reset
</details>
Fixed

---

*[libs/drivers/antenna/antenna.c, line 113 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSeqWF6WAgqc93wnjS:-KWntZ3nMuPUK4FKsfp7:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L113)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Missing `static`
</details>
Fixed

---

*[libs/drivers/antenna/antenna.c, line 144 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSf58CiSHQXDT51b5F:-KWnte368rqIkUAC0F2P:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L144)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Missing `static`
</details>
Fixed

---

*[libs/drivers/antenna/antenna.c, line 243 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSeIYZz-oGcToWrtCj:-KWntfHIdxpgr6HVPtQH:b-l80lnz) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/antenna.c#L243)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Why always checking both controllers even if primary is working?
</details>
there are two temperature sensors each on each controller, I read both of them all to simplify interface as opposed to the necessity to require user to access it separately for each channel.

---

*[libs/drivers/antenna/miniport.c, line 22 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS_njzYoYc02Vzxa6B:-KWnujCEMYxSqp1z2Y_a:b-pmu1ql) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L22)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Why are some enum names CamelCase (like OSResult) but this is upper case with underscores without any prefix?
</details>
public enums (defined in header files) are all Upper case as our coding standard defines. I define private enumerators  using CamelCase.

---

*[libs/drivers/antenna/miniport.c, line 54 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSZlfdoR5ck9b98YK-:-KWoDkfQBvCX2FWVkV6W:b-1k24zv) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L54)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> I see some repeating pattern of sending commands without parameters with optional response. Just thinking maybe is useful to extract it to some common utility functions?
</details>
That may be useful but I would wait until we get more use cases 

---

*[libs/drivers/antenna/miniport.c, line 199 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSaoDy09S8NSwoX4_w:-KWoE44m000XY1cW9Ceu:bb6un1j) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L199)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> AVR-Libc defines few useful macros for bit operations: `bit_is_set` and `bit_is_clear`. I would say it will make code more readable if we include these functions in our codebase 
> 
> Source: http://www.nongnu.org/avr-libc/user-manual/sfr__defs_8h_source.html#234
</details>
Done

---

*[libs/drivers/antenna/Include/antenna/miniport.h, line 21 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWcTLGfIUeI4QPAa3gC:-KWoE82pJdaQZkjRrH8L:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/Include/antenna/miniport.h#L21)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> **a**t this level
</details>
Fixed

---

*[libs/drivers/antenna/Include/antenna/miniport.h, line 30 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWcTRvyKrXsTuCTkPW0:-KWoEKEyxjMuULLiER0f:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/Include/antenna/miniport.h#L30)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> even**t** concurrently - that's not exaclty true. You can't send two commands at the same time on two channels if they are using the same underlying hardware bus. 
</details>
Fixed

---

*[src/commands/antenna.c, line 63 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWZg0mdJk7TOxMu2y_G:-KWoFHeFAdwghvmxHNGS:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/src/commands/antenna.c#L63)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Move it to utils
</details>
Fixed

---

*[unit_tests/antenna/miniport.cpp, line 133 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdSHWhx2cw9wJaPgGB:-KWoFJ5LmOa2DkNRfeb-:b7f5i27) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/unit_tests/antenna/miniport.cpp#L133)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> (Just thinking) how about creating some matchers dedicated for I2C transfers?
</details>
It is worth considering. Maybe I will define them next time...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 1 of 10 files at r4, 14 of 14 files at r6.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[integration_tests/devices/antenna.py, line 200 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSVeVUaz3wqAfiSMNq:-KWqPyKINwlWnrU8UHs4:b-jz1jk2) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L200)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary>
> It may work but I still prefer explicit function call.
</details>
I was thinking about:
```
flags = [
      not self.antenna_state[0].is_deployed(),
      self.antenna_state[0].is_being_deployed()
]

num = reduce(lambda s,b: (s << 1) | b, flags, 0)
```
bu I will not fight for it ;)

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWqQT_hptmxcrAWPgiI:b-hh8hm7) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary>
> Fine
</details>
What about first part? The `_` prefix?

---

*[integration_tests/tests/__init__.py, line 14 at r6](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWqQxZN657FCORccCA7:-KWqQxZO2-c0FkL2Il1m:b-9k4vmz) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/0ffcb080f603ab4e6a4540e95675865d0dac846f/integration_tests/tests/__init__.py#L14)):*
> ```Python
> 
> def _setup_log():
>     logging.basicConfig()
> ```

Whoa. Code below is already setting default logging config. There's no need to configure it again

---

*[integration_tests/tests/test_antenna.py, line 97 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdUCiqasG0Ku3CCKNa:-KWqRNE6oXf8K33x_zAp:b-sfe2pi) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L97)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary>
> what would you put in such message that is not already expressed in the code?
</details>
it's only about reading logs in case of tests error, all you get is 'Expected True actual: False' and you have to dig into code to find issue. I want meaningful messages in tests like 'Expected antenna 1 deployment to be in progress'

---

*[libs/drivers/antenna/miniport.c, line 22 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWS_njzYoYc02Vzxa6B:-KWqSyEFNT50KEZRWtmp:b-vpmxn4) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/libs/drivers/antenna/miniport.c#L22)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary>
> public enums (defined in header files) are all Upper case as our coding standard defines. I define private enumerators  using CamelCase.
</details>
So most of out enums are not following that standard ;) Let's leave it for now and revise naming style of enums 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Reviewed 1 of 20 files at r3, 1 of 14 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[integration_tests/devices/antenna.py, line 200 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSVeVUaz3wqAfiSMNq:-KWqXicsX--hqcRgE3ty:b-7y0d9z) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L200)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> I was thinking about:
> ```
> flags = [
>       not self.antenna_state[0].is_deployed(),
>       self.antenna_state[0].is_being_deployed()
> ]
> 
> num = reduce(lambda s,b: (s << 1) | b, flags, 0)
> ```
> bu I will not fight for it ;)
> 
</details>
So let's leave it as it is.

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWqXkQjzgVXBTTJvu5h:b-v2xmgn) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> What about first part? The `_` prefix?
</details>
All of the non command methods are now prefixed with underscore.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWqZGE6aEGWurrWSFBs:b-mf85cy) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
<details><summary><i>Previously, zaq32 (Łukasz Żak) wrote…</i></summary>
> All of the non command methods are now prefixed with underscore.
</details>
I though about prefix all i2c command methods with `_`, like ` cancel_deployment` -> `_ cancel_deployment`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[integration_tests/devices/antenna.py, line 226 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWSW2sMB35JJVbg6Qj3:-KWt0KgGFkuuLS8uzx7h:b499q7j) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/devices/antenna.py#L226)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> I though about prefix all i2c command methods with `_`, like ` cancel_deployment` -> `_ cancel_deployment`
</details>
Changed

---

*[integration_tests/tests/__init__.py, line 14 at r6](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWqQxZN657FCORccCA7:-KWt6QlwjCZUH-otWl1t:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/0ffcb080f603ab4e6a4540e95675865d0dac846f/integration_tests/tests/__init__.py#L14)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> Whoa. Code below is already setting default logging config. There's no need to configure it again
</details>
Fixed

---

*[integration_tests/tests/test_antenna.py, line 23 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdTZmz9hUz-gxPRWcG:-KWt2UZHWVaPBa887Aq3:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L23)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> I'd prefer to have this as decorator `@auto_power_on(False)`
</details>
Fixed

---

*[integration_tests/tests/test_antenna.py, line 97 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWdUCiqasG0Ku3CCKNa:-KWt2vrlN3rTTMogR1b0:bb6vskf) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/tests/test_antenna.py#L97)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> it's only about reading logs in case of tests error, all you get is 'Expected True actual: False' and you have to dig into code to find issue. I want meaningful messages in tests like 'Expected antenna 1 deployment to be in progress'
</details>
Fine

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 6 of 6 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

Novakov commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-:-KXCEXh6l3NOnYuFBFv1:bnfp4nl)*
<!-- Sent from Reviewable.io -->

Novakov commented:
Reviewed 27 of 35 files at r8.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/CMakeLists.txt, line 13 at r8](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KXHfp9dBL_p7rMW-euW:-KXHfp9dBL_p7rMW-euX:bq21yjh) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4f7be1c27af2e3e2ba11a0e1a35b00bd36f2df90/src/CMakeLists.txt#L13)):*
> ```CMake
>     commands/state_command.cpp
>     commands/i2c_test_command.cpp
> 	commands/antenna.cpp
> ```

The same file is listed twice

---

*[src/main.cpp, line 290 at r8](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KXHg7DSf83H8kFvojQX:-KXHg7DTjDErRW9LcoXI:btyaj8t) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4f7be1c27af2e3e2ba11a0e1a35b00bd36f2df90/src/main.cpp#L290)):*
> ```cpp
> 
>     Main.I2C.Bus = (I2CBus*)&Main.I2CBuses[I2C_SYSTEM_BUS].ErrorHandling;
>     Main.I2C.Payload = (I2CBus*)&Main.I2CBuses[I2C_PAYLOAD_BUS].ErrorHandling;
> ```

This has been moved to OBC ctor. No need to do it here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

zaq32 commented:
Review status: 46 of 52 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/CMakeLists.txt, line 13 at r8](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KXHfp9dBL_p7rMW-euW:-KXKAhdMuqp_MG1npSnM:b-886gsx) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4f7be1c27af2e3e2ba11a0e1a35b00bd36f2df90/src/CMakeLists.txt#L13)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> The same file is listed twice
</details>
Fixed

---

*[src/main.cpp, line 290 at r8](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KXHg7DSf83H8kFvojQX:-KXKCXe76pRx7I9NZnPw:bt2vr1b) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/4f7be1c27af2e3e2ba11a0e1a35b00bd36f2df90/src/main.cpp#L290)):*
<details><summary><i>Previously, Novakov (Maciej T. Nowak) wrote…</i></summary>
> This has been moved to OBC ctor. No need to do it here
</details>
Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

Novakov commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 8 of 9 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-:-KXMXla9468PNnNy5IE2:bnfp4nl)*
<!-- Sent from Reviewable.io -->

decu69 commented:
*[integration_tests/devices/antenna.py, line 5 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWJRijaJXiKTE8qEC4y:-KWJRijb9GTzTg3_QHUw:bt921b9) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/antenna.py#L5)):*
> ```Python
> import i2cMock
> from i2cMock import I2CDevice
> import inspect
> ```

this import should be higher than import i2cMock as it more general package

from: https://www.python.org/dev/peps/pep-0008/#imports

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->

decu69 commented:
Reviewed 5 of 37 files at r1, 5 of 13 files at r2, 4 of 23 files at r3, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[integration_tests/devices/__init__.py, line 5 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWJPMuSkI33exs-9lBP:-KWJPMuSkI33exs-9lBQ:btkmq6o) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/__init__.py#L5)):*
> ```Python
> from antenna import *
> from antenna import PRIMARY_ANTENNA_CONTROLLER_ADDRESS
> from antenna import BACKUP_ANTENNA_CONTROLLER_ADDRESS
> ```

so implicit or explicit import from anttenna?

---

*[integration_tests/devices/antenna.py, line 7 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWJTFeUw89QKwzt_w23:-KWJTFeUw89QKwzt_w24:b-xziuwe) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/antenna.py#L7)):*
> ```Python
> import inspect
> 
> class Antenna():
> ```

I believe it would be better to use new-style classes inherited from object. Old-ones are deprecated. 

http://stackoverflow.com/questions/54867/what-is-the-difference-between-old-style-and-new-style-classes-in-python

---

*[integration_tests/devices/antenna.py, line 12 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWJTc4ZYHmvNFL9XCTB:-KWJTc4ZYHmvNFL9XCTC:b-uekdpt) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/antenna.py#L12)):*
> ```Python
>     is_being_deployed = False
> 
>     def is_deployed(self):
> ```

@property?

---

*[integration_tests/devices/antenna.py, line 26 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWJU7K9CXAKXsqra7_U:-KWJU7KA7z1yiSYoDt1Q:b-w8xfmh) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/antenna.py#L26)):*
> ```Python
>         self.is_being_deployed = False
> 
> PRIMARY_ANTENNA_CONTROLLER_ADDRESS = 0x32
> ```

"Constants are usually defined on a module level and written in all capital letters with underscores separating words. "

---

*[integration_tests/devices/antenna.py, line 54 at r3](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWO1SB2ZeK_XSRLQ-a8:-KWO1SB2ZeK_XSRLQ-a9:b-j7pc6l) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/28c465b8ca971df8aa764e95b80433f27bee9056/integration_tests/devices/antenna.py#L54)):*
> ```Python
>     @staticmethod
>     def call(method, default, *args):
>         if not method is None:
> ```

isnt that always true? 
if method ---- would not work?
or "if method is not None"?

I think I might not understand what do you check here.

---

*[integration_tests/obc/antenna.py, line 2 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWOq8z1IQb7rIRMl72j:-KWOq8z2CZaDELg328yV:b-9fkd6x) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/obc/antenna.py#L2)):*
> ```Python
> from .obc_mixin import OBCMixin, command, decode_return
> from enum import Enum, unique
> ```

imports order

---

*[integration_tests/system/system.py, line 43 at r5](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40#-KWOrCxcGxlX_E5rSrrG:-KWOrCxcGxlX_E5rSrrH:b-hutc1j) ([raw file](https://github.com/pw-sat2/pwsat2obc/blob/48a3fd42571d66c79c35d75fabcbd5b232ff7b0c/integration_tests/system/system.py#L43)):*
> ```Python
>         self.receiver = ReceiverDevice()
>         self.primary_antenna = AntennaController(PRIMARY_ANTENNA_CONTROLLER_ADDRESS)
>         self.backup_antenna = AntennaController(BACKUP_ANTENNA_CONTROLLER_ADDRESS)
> ```

backup address?  why do we need backup antenna?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pw-sat2/pwsat2obc/40)*
<!-- Sent from Reviewable.io -->
